packages/ember-cli-eyeglass/src/broccoli-ln-s.ts

```diff
@@ -0,0 +1,82 @@
+import BroccoliPlugin = require("broccoli-plugin");
+import path = require("path");
+import * as fs from "fs-extra";
+
+type EnsureSymlinkSync = (srcFile: string, destLink: string) => void;
+/* eslint-disable-next-line @typescript-eslint/no-var-requires */
+const ensureSymlink: EnsureSymlinkSync = require("ensure-symlink");
+
+/**
+ * An object where the keys are the files that will be created in the tree and
+ * the values are the source files.
+ * 
+ * @interface FileMap
+ */
+interface FileMap {
+  [relativePath: string]: string;
+}
+
+type BroccoliSymbolicLinkerOptions =
+  Pick<BroccoliPlugin.BroccoliPluginOptions, "annotation" | "persistentOutput">;
+
+/**
+ * Creates symlinks to the source files specified.
+ * 
+ *  BroccoliSymbolicLinker
+ */
+export class BroccoliSymbolicLinker extends BroccoliPlugin {
```

Comment:
this plugin likely needs to be tested, https://github.com/broccolijs/broccoli-test-helper can be quite helpful for that.